release: Add script to publish images on github

jcvenegas opened the pull request:
This PRs add automation to build a new image based on runtime versions file and publish a new image on github.com/osbuilder/releases.

jodh-intel commented on clear-containers-image/release_image_github.sh:
```diff
@@ -0,0 +1,166 @@
+versions_file="https://raw.githubusercontent.com/clearcontainers/runtime/master/versions.txt"
+os_version="$(curl -Ls ${versions_file} | grep '^clear_vm_image_version=' | cut -d= -f 2)"
+agent_version="$(curl -Ls ${versions_file} | grep '^cc_agent_version=' | cut -d= -f 2)"
+#Fixme: remove after https://github.com/clearcontainers/runtime/pull/923 is merged
```
I've just merged this change, so can you remove this comment and adjust `agent_version=` below?

jodh-intel commented on clear-containers-image/release_image_github.sh:
```diff
@@ -0,0 +1,166 @@
+git clone "https://github.com/${repo_owner}/osbuilder.git" "${osbuilder_dir}"
+pushd osbuilder
+
+#FIXME:
```
Can this chunk be removed now https://github.com/clearcontainers/runtime/pull/923 is merged?

jodh-intel reviewed (changes requested):
A couple of minor (expected) changes.

I haven't forgotten about updating checkcommits btw - will try to get that resolved this week...

jcvenegas committed:
release-tools: Add script to publish image on github.

This commit adds a new scipt release_image_github.sh

The script will create a new image based on
content from versions.txt.

It will use the Clear Linux version and
agent version from versions.txt

Then will create an github release on osbuilder.

Fixes: #243

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas commented:
Image generated by this script:
https://github.com/clearcontainers/osbuilder/releases/tag/cc-19350-agent-b1d92e

ci using new image passing:
https://github.com/clearcontainers/tests/pull/839#partial-pull-merging

devimc commented on release-tools/release_image_github.sh:
```diff
@@ -0,0 +1,158 @@
+-o <dir>        : Create resulting image in <dir>
+-p              : Push to git-repo default: ${push}
+-r <repo-owner> : git repository to push image
+-t <token>      : Github token to create new release. ENV: \$GITHUB_TOKEN
```
which one have more priority?

jcvenegas replied:
the -t options has more priority, added usage.

jcvenegas commented:
@jodh-intel  @devimc  @erick0z  please review again, ready to merge.

devimc commented on release-tools/release_image_github.sh:
```diff
@@ -0,0 +1,159 @@
+-p              : Push to git-repo default: ${push}
+-r <repo-owner> : git repository to push image
+-t <token>      : Github token to create new release. ENV: \$GITHUB_TOKEN
+                  this option has higher priority than env variables
```
this option takes priority over environment variable

jcvenegas replied:
Fixed,

devimc commented:
lgtm

jodh-intel commented on release-tools/Makefile.dist.install.image:
```diff
@@ -0,0 +1,29 @@
+
+install:
+	install -D --owner root --group root --mode 0644 $(IMAGE) $(IMAGE_DEST)
+	ln -sf $(IMAGE_DEST) $(DESTSYSCONFDIR)/clear-containers.img
```
Should this be?:

```
ln -sf $(IMAGE_DEST) $(IMAGE)
```

jcvenegas replied:
No, it should be to comply with our current symlink pattern to point latest installed image to DESTSYSCONFDIR/clear-containers.img

jodh-intel replied:
ok. I'd create a top-level variable for this though to make it more obvious how to switch it if it goes into kata.

jcvenegas replied:
Added variable

jodh-intel commented on release-tools/release_image_github.sh:
```diff
@@ -0,0 +1,223 @@
+	image_name="${image_version}.img"
+	mkdir -p "${dist_bin_name}"
+
+	cp workdir/container.img "${dist_bin_name}/${image_name}"
```
- Can you avoid hard-coding `workdir` - where is that coming from btw?
- Also, can you add a variable for `container.img`?

jcvenegas replied:
that is from osbuilder, because I did not changed the workdir in from osbuilder it is creating the image in the current current working directory, yep I will change make this explicit.

jcvenegas replied:
Added a variable to make it explicit.

jodh-intel commented on release-tools/release_image_github.sh:
```diff
@@ -0,0 +1,223 @@
+		;;
+
+	check-updated)
+		tarbal_url=$(get_image_tarball_url "${agent_version}" "${os_version}")
```
Nit: `tarball_url` here and below.

jodh-intel commented on release-tools/release_image_github.sh:
```diff
@@ -0,0 +1,223 @@
+	check-updated)
+		tarbal_url=$(get_image_tarball_url "${agent_version}" "${os_version}")
+		if curl -o /dev/null --silent --head --fail "$tarbal_url"; then
+			echo "Image uptodate: $tarbal_url"
```
"up-to-date"

jcvenegas commented:
@jodh-intel changes applied

jodh-intel commented:
Thanks @jcvenegas !
